src/controller/CHIPDeviceController.cpp

```diff
@@ -736,6 +744,7 @@ CHIP_ERROR DeviceCommissioner::EstablishPASEConnection(NodeId remoteDeviceId, Re
                     // We already have an open secure session to this device, call the callback immediately and early return.
                     mPairingDelegate->OnPairingComplete(CHIP_NO_ERROR);
                 }
+                MATTER_LOG_METRIC_END(kMetricDeviceCommissionerPASESession, CHIP_NO_ERROR);
                 return CHIP_NO_ERROR;
             }
             if (current->IsSessionSetupInProgress())
```

Comment:
Do we need an end metric inside this block too?

Or perhaps: should we only have the "start" metric when we get around to actually doing the `Pair()` call?  I don't quite understand what the `kMetricDeviceCommissionerPASESession` represents given the code here right now.

Reply:
Actually I don't think we want to end here. As the comment says, the delegate will be called and that is where the end of the PASE session will be marked

Reply:
Agree